sky/task.py

```diff
@@ -903,6 +904,21 @@ def update_storage_mounts(
         task_storage_mounts.update(storage_mounts)
         return self.set_storage_mounts(task_storage_mounts)
 
+    def get_bucket_name_and_store_type_from_job_config(
+            self) -> Tuple[Optional[str], Optional[str]]:
+        """Returns the bucket name and store type from the job config."""
+        bucket_dict = skypilot_config.get_nested(('jobs', 'bucket'), None)
+        if bucket_dict is None:
+            return None, None
+        store_type, _ = self._get_preferred_store()
+        if store_type.value.lower() in bucket_dict:
+            bucket_name = bucket_dict[store_type.value.lower()]
+        elif 'default' in bucket_dict:
+            bucket_name = bucket_dict['default']
+        else:
+            return None, None
+        return bucket_name, store_type.value
```

Comment:
This should probably not be a task method, since this is highly specific to jobs controller and does not apply to regular tasks.